dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -221,4 +230,75 @@ public KernelBuilder Configure(Action<KernelConfig> configure)
         this._aiServices.SetService<TService>(serviceId, () => factory(this._loggerFactory, this._config), setAsDefault);
         return this;
     }
+
+    /// <summary>
+    /// Create a default prompt template engine.
+    ///
+    /// This is a temporary solution to avoid breaking existing clients.
+    /// There will be a separate task to add support for registering instances of IPromptTemplateEngine and obsoleting the current approach.
+    ///
+    /// </summary>
+    /// <param name="loggerFactory">Logger factory to be used by the template engine</param>
+    /// <returns></returns>
+    private IPromptTemplateEngine CreateDefaultPromptTemplateEngine(ILoggerFactory? loggerFactory = null)
```

Comment:
Consider moving this logic to a `NullTemplateEngine` static `Instance`, and call it from the Builder `this._promptTemplateEngine ?? NullTemplateEngine.Instance`

Reply:
I agree, we should not keep `PromptTemplateEngine`-related logic in `KernelBuilder`. Regarding `this._promptTemplateEngine ?? NullTemplateEngine.Instance`, we can put this logic in place where template engine is actually used, in similar way how we do this with logging.

Reply:
This is deliberate. The goal with this PR is just to move the code and not cause a breaking change. We need to decide on a pattern for how to remove the logic from the `KernelBuilder` for the template engine (and also memory) but that is out of scope for this PR.

Reply:
You can create separate class `DefaultPromptTemplateEngine` temporarily, place it in separate file and move all the logic there. You will have to do the same for `NoopPromptTemplateEngine` in order not to keep two classes in same file.